### PR TITLE
ComboboxList layout props

### PR DESCRIPTION
**Pull request by mdodgelooker**
### :sparkles: Changes

- Added `LayoutProps` to `ComboboxList`
- Added `listLayout` prop to `Select` & `SelectMulti` to enable access to the above

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [x] Checked for i18n impacts
- [x] Checked for a11y impacts
- [x] PR is ideally < ~400LOC

### :camera: Screenshots
image

### CODE REVIEW

File: packages/components/src/Form/Inputs/Combobox/ComboboxContext.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export interface ComboboxContextProps<
   inputCallbackRef?: Ref<HTMLInputElement>
   inputElement?: HTMLInputElement | null
   wrapperElement?: HTMLDivElement | null
-  popoverRef?: RefObject<HTMLDivElement>
+  listRef?: RefObject<HTMLUListElement>
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Combobox/ComboboxContext.ts:58]: After removing `PopoverContent` from `ComboboxList`, the name `popoverRef` felt wrong.

File: packages/components/src/Form/Inputs/Combobox/ComboboxList.tsx [hunk 2]
```diff
@@ -138,26 +142,23 @@ const ComboboxListInternal = forwardRef(
...
-      : 'auto'
+    const ref = useForkedRef(listRef, forwardedRef)
+
+    const width =
+      props.width ||
+      (wrapperElement && wrapperElement.getBoundingClientRect().width) ||
+      'auto'
 
     const content = (
-      <PopoverContent
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Combobox/ComboboxList.tsx:146]: `PopoverContent` is totally superfluous here – not sure why it took me this long to realize.

File: packages/components/src/Form/Inputs/Combobox/ComboboxList.tsx [hunk 3]
```diff
@@ -138,26 +142,23 @@ const ComboboxListInternal = forwardRef(
...
         onBlur={handleBlur}
-        ref={popoverRef}
-        p="none"
-      >
-        <ul
-          {...props}
-          ref={forwardedRef}
-          role="listbox"
-          tabIndex={-1}
-          data-testid="combobox-list"
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Combobox/ComboboxList.tsx:158]: Not needed – `getByRole('listbox')` can be used instead.

File: packages/components/src/Form/Inputs/Combobox/ComboboxList.tsx [hunk 4]
```diff
@@ -221,28 +222,28 @@ const ComboboxListInternal = forwardRef(
...
-const comboboxListStyles = css<ComboboxListInternalProps>`
+const ComboboxUl = styled.ul<ComboboxListProps>`
   ${reset}
   ${typography}
   ${space}
   list-style-type: none;
   margin: 0;
-  padding: ${({ isMulti, theme }) => (isMulti ? theme.space.xsmall : 0)} 0;
   max-height: 30rem;
+  ${layout}
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Combobox/ComboboxList.tsx:232]: Putting `${layout}` after `max-height: 30rem;` so the developer can override it if they want.

File: packages/components/src/Form/Inputs/Combobox/utils/useKeyDown.ts [hunk 5]
```diff
@@ -81,7 +81,9 @@ export function useKeyDown() {
     }
   }
 
-  return throttle(function handleKeyDown(event: KeyboardEvent<HTMLDivElement>) {
+  return throttle(function handleKeyDown(
+    event: KeyboardEvent<HTMLUListElement | HTMLDivElement>
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Combobox/utils/useKeyDown.ts:85]: The listener is added to both the list (which is no longer wrapped in a div) and the input.

File: packages/components/src/Form/Inputs/Combobox/utils/useKeyDown.ts [hunk 6]
```diff
@@ -81,7 +81,9 @@ export function useKeyDown() {
     }
   }
 
-  return throttle(function handleKeyDown(event: KeyboardEvent<HTMLDivElement>) {
+  return throttle(function handleKeyDown(
+    event: KeyboardEvent<HTMLUListElement | HTMLInputElement>
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Combobox/utils/useKeyDown.ts:85]: This listener is added to both the list, which is no longer wrapped in a div, and the input.

File: packages/components/src/Form/Inputs/Combobox/Combobox.test.tsx [hunk 7]
```diff
@@ -149,6 +149,50 @@ describe('<Combobox/> with children', () => {
...
+    const { getByRole, getByPlaceholderText } = renderWithTheme(jsx)
+
+    const input = getByPlaceholderText('Type here')
+    fireEvent.click(input)
+
+    const list = getByRole('listbox')
+    expect(list).toHaveStyleRule('max-height', '400px')
+    expect(list).toHaveStyleRule('max-width', '800px')
+    expect(list).toHaveStyleRule('min-width', '300px')
+    expect(list).toHaveStyleRule('width', '50vw')
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/Combobox/Combobox.test.tsx:190]: `toHaveStyleRule` is a great way to test specifics without having to snapshot the whole thing 👍

**Pull request merged**